x-pack/plugins/canvas/canvas_plugin_src/functions/server/src/esdocs.js

```diff
@@ -101,6 +101,16 @@ export const esdocs = () => ({
           columns,
           rows,
         };
+      })
+      .catch(e => {
+        if (e.message.indexOf('parsing_exception') > -1) {
+          throw new Error(
+            `Couldn't parse Elasticsearch SQL query. You may need to add double quotes to names containing special characters. Check your query and try again. Error: ${
```

Comment:
nit: seeing an "Elasticsearch SQL" error when you're using `esdocs` seems a little confusing, but may not be a problem since the error does tell you what function is generating the error.